### PR TITLE
ckan 2.10.4

### PR DESCRIPTION
**Pull request by FuhuXia**
# Pull Request

Related to
- https://github.com/GSA/data.gov/issues/4559

## About
- update core core base image
- update requirement list
- use actual pub cert for saml2, not placeholder

## PR TASKS

- [x] The actual code changes.
- [x] Tests written and passed.
- [ ] Any changes to docs?
- [ ] Any new
[requirements versions](https://github.com/GSA/inventory-app/blob/main/requirements.in.txt)
to pull in?

### CODE REVIEW

**Comment by FuhuXia** [comment]
deployed to develop space.  saml login is working. `status_show` shows 2.10.4. 

https://inventory-dev-datagov.app.cloud.gov/api/action/status_show

**Review by robert-bryson — APPROVED** [review]

File: ckanext/datagov_inventory/tests/logic/auth/test_auth.py [hunk 1]
```diff
@@ -378,11 +378,11 @@ def test_auth_user_list(self):
         self.setup_test_orgs_users()
 
         self.assert_user_authorization('user_list', {
-            'gsa_admin': is_denied,
-            'gsa_editor': is_denied,
-            'gsa_member': is_denied,
-            'doi_admin': is_denied,
-            'doi_member': is_denied,
+            'gsa_admin': is_allowed,
```
Comment (btylerburton) [inline ckanext/datagov_inventory/tests/logic/auth/test_auth.py:381]: why did these change?

Comment (FuhuXia) [reply ckanext/datagov_inventory/tests/logic/auth/test_auth.py:381]: No one actually knows. I remember when the tests were written, it was more like describing what is happening, not testing what should be happening.

Comment (FuhuXia) [reply ckanext/datagov_inventory/tests/logic/auth/test_auth.py:381]: Checked on inventory with ckan 2.10.1 and ckan 2.10.4. The new test result matches the actual behavior on both. So maybe previous test was wrong in 2.10.1 and it got fixed in the new version2.10.4.

**Pull request merged**